tests/unit/query_plan.cpp

```diff
@@ -1685,5 +1709,29 @@ TYPED_TEST(TestPlanner, Foreach) {
         QUERY(SINGLE_QUERY(FOREACH(i, {CREATE(PATTERN(NODE("n")))}), FOREACH(j, {CREATE(PATTERN(NODE("n")))})));
     CheckPlan<TypeParam>(query, storage, ExpectForeach(input, updates), ExpectEmptyResult());
   }
+
+  {
+    // FOREACH with index
```

Comment:
Here we just verify that the planner does visit the inside of the `FOREACH` clause right? And I assume (based on the changes I saw) that didn't work before, right?

Reply:
yes. It used the ScanAll index rather than indexed scan.